blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/Caches.java

```diff
@@ -95,10 +100,24 @@ public void onDeleted(Item item) {
             }
             ObjectMetadataAction om = job.getAction(ObjectMetadataAction.class);
             PrimaryInstanceMetadataAction pima = job.getAction(PrimaryInstanceMetadataAction.class);
-            if (om == null && pima == null) {
+            String url = om != null && om.getObjectUrl() != null ? om.getObjectUrl() : null;
+            if (StringUtils.isEmpty(url)) {
+                /*
+                 * Borrowed from https://github.com/jenkinsci/branch-api-plugin/blob/c4d394415cf25b6890855a08360119313f1330d2/src/main/java/jenkins/branch/BranchNameContributor.java#L63
+                 * for those that don't implement object metadata action
+                 */
+                ItemGroup parent = job.getParent();
+                if (parent instanceof MultiBranchProject) {
+                    BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
+                    if (projectFactory.isProject(job)) {
+                        SCMHead head = projectFactory.getBranch(job).getHead();
+                        url = head.getName();
+                    }
+                }
+            }
+            if (StringUtils.isEmpty(url) && pima == null) {
```

Comment:
Why do  we need to check again for `isEmpty(url)` ? Couldn't you also add an `} else if(pima == null) {` on line 115 above?

Other then that the logic looks fine. Just curious why the additional check.

Reply:
Its a long while ago that I wrote this, but I'm pretty sure its because I assumed head.getName() wasn't a) guaranteed to be returned, and b) guaranteed to get there.

If url is set to something, we can not even bother to check pima.